core/src/main/java/com/amplifyframework/api/graphql/PaginatedResult.java

```diff
@@ -15,15 +15,13 @@
 
 package com.amplifyframework.api.graphql;
 
-import com.amplifyframework.core.model.Model;
-
 /**
  * Represents a page of results returned from an API.  Specifically, contains the list of items in the page, as well as
  * a GraphQLRequest which can be used to obtain the next page.
  *
- * @param <T> Type of the items in the list.  Must extend Model.
+ * @param <T> Type of the items in the list.
  */
-public abstract class PaginatedResult<T extends Model> {
```

Comment:
Hm. Are there any remaining types we want/need to paginate over, that are not models?

`Iterable<String>`, `<String>`, `ModelWithMutation<T>`, `CustomGizmoNotFromCodeGen`?

Reply:
The reason I changed this from <T extends Model> to <T> is so we can paginate over ModelWithMetadata<M>, since ModelWithMetadata is not a Model itself.

Reply:
Maybe it can be `PaginatedResult<ModelWithMetadata<T extends Model>>`?

Reply:
> Maybe it can be `PaginatedResult<ModelWithMetadata<T extends Model>>`?

That would work for DataStore, but not for API, because API expects a PaginatedResult<? extends Model>.

Reply:
Ah...makes sense. I guess the only other alternative I can think of would be to make ModelWithMetadata a Model.

```
public final class ModelWithMetadata<M extends Model> implements Model {
     private final M model;

    @NonNull
    @Override
    public String getId() {
        return model.getId();
    }
}
```

Not sure if it's worth it though unless it would simplify all the other bits of code.

Reply:
Yeah, it's not really a `Model` though.  We could create a "PaginatedResultItem` interface, which is implemented by `Model` and `ModelWithMetadata` though.  Not sure if that's really worth it right now, but I'll think about that for my next PR, which is going to get a little deeper into this code.